gradio/components.py

```diff
@@ -1107,15 +1136,21 @@ def preprocess(self, x: str) -> str | int:
         Parameters:
             x: selected choice
         Returns:
-            selected choice as string or index within choice list
+            - if type is "value" and x is not in "choices" list raise ValueError otherwise return x. Note if x is None we return the element at the first index in the "choices" list.
```

Comment:
```suggestion
            - if type is "value" and x is not in `self.choices` list raise ValueError otherwise return x. Note if x is None we return the element at the first index in the "choices" list.
```